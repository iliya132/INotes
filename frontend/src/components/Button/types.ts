import { MouseEventHandler } from "react";

export interface IButtonProps {
    title: string,
    className?: string,
    disabled?: boolean;
    onClick?: MouseEventHandler<HTMLButtonElement>;
}