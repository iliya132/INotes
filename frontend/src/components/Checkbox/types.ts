/* eslint-disable no-unused-vars */
export interface ICheckboxProps {
    defaultValue?: boolean;
    id: string;
    className?: string;
    onChanged? : (newValue:boolean) => void;
    title?: string;
}
