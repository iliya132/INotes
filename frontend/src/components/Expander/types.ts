export interface IExpanderProps {
    isExpanded: boolean;
    expanderBody: JSX.Element;
    children: JSX.Element;
    onClick?: () => void;
}
