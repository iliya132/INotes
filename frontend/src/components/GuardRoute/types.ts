export interface IGuardRouteProps {
    canActivate: boolean;
    redirectTo: string;
    children: JSX.Element;
    onSuccess?: () => void
  }
  