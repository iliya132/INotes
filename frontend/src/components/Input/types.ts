import React, { HTMLInputTypeAttribute } from "react";
import { Icons } from "../Svg/types";

export default interface IInputProps{
    type: HTMLInputTypeAttribute;
    id: string;
    autocomplete?: 'username' | 'email' | 'password' | 'new-password' | 'current-password' | 'disabled';
    error?: string,
    className?: string,
    onChange?: React.ChangeEventHandler<HTMLInputElement>,
    icon?: Icons,
    placeholder?: string,
    defaultValue?: string
}
