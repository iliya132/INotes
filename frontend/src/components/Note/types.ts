/* eslint-disable no-unused-vars */
import { INote } from "../../store/types";

export interface INoteProps{
    note: INote;
    onClick?: (note: INote) => void;
    isSelected?: boolean;
}
