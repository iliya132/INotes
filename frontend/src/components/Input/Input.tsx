import React, { useState } from 'react';
import IInputProps from './types';
import styles from './input.scss';
import Svg from '../Svg';
import { Icons } from '../Svg/types';
import classNames from 'classnames';

export function Input(props: IInputProps) {
    const { type, id, autocomplete, error, className, onChange } = props;
    const [isPasswordShown, setPasswordShown] = useState(false);
    const inputType = type === 'password' && isPasswordShown ? 'text' : type;
    const internalClassName = error ? classNames(styles['input'], styles['input-error'], className) : classNames(styles['input'], className);
    return (
        <>
            <div className={styles['input-container']}>
                <input className={internalClassName} id={id} type={inputType} onChange={onChange} autoComplete={autocomplete}/>
                {inputType == 'password' ? (
                    <Svg
                        icon={Icons.EyeOpened}
                        className={styles['eye-icon']}
                        onClick={() => setPasswordShown(!isPasswordShown)}
                    />
                ) : type === 'password' ? (
                    <Svg
                        icon={Icons.EyeClosed}
                        className={styles['eye-icon']}
                        onClick={() => setPasswordShown(!isPasswordShown)}
                    />
                ) : null}
            </div>
            <label className={styles['error-label']} htmlFor={id}>{error}</label>
        </>
    );
}
