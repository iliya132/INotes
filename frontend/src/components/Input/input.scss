@import '../../Misc/colors.scss';

.input-container {
    width: 100%;
    display: grid;
    height: 30px;
    grid-template-columns: 1fr 50px;
    border-bottom: 1px solid $TextColor;
}

.input-container .input {
    width: 100%;
    height: 30px;
    background-color: transparent;
    border: none;

    font-size: 13px;
    color: $TextColor;
    line-height: 13px;
    outline: none;
}

.eye-icon {
    width: 48px;
    height: 48px;
    position:relative;
    grid-column: 2;
    margin-top: -8px;
    cursor: pointer;
}

.error-label{
    color: $Error;
    font-size: 9px;
    font-weight: 400;
}

.input-error{
    color:$Error !important;
}
