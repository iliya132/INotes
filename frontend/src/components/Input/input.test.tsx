/* eslint-disable no-undef */
import React from 'react';
import { createRoot, Root } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import { Icons } from '../Svg/types';
import { Input } from './Input';

//#region common setup
let root: Root | null = null;
let container: Element | null = null;
beforeEach(() => {
    act(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        root = createRoot(container);
    });
});

afterEach(() => {
    act(() => {
        root!.unmount();
        container!.remove();
        container = null;
    });
});
//#endregion

describe("test  input component", () => {
    it("matches snapshot", () => {
        act(() => {
            root!.render(<Input id='test-id' type="text" icon={Icons.Key} />)
        })
        expect(container!.firstChild).toMatchSnapshot("Input test snapshot")
    })

    it("should render icon when provided", () => {
        act(() => {
            root!.render(<Input id='test-id' type="text" icon={Icons.Key} />)
        })
        const svg = container!.querySelector("svg")
        expect(svg).not.toBeNull()
        expect(svg?.outerHTML).toContain("#key")
    })

    it("should show error when provided", () =>{
        act(()=> {
            root!.render(<Input id="test-id" error='test-error' type="text"/>)
        })

        const expectedInputClassName = "input input-error"
        const errorLabel = container!.querySelector("label")
        expect(errorLabel?.textContent).toEqual("test-error")

        const input = container!.querySelector("input")
        expect(input?.className).toEqual(expectedInputClassName)
    })
})