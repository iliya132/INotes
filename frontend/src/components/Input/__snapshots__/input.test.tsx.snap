// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  input component matches snapshot: Input test snapshot 1`] = `
<div
  class="input-container"
>
  <svg
    class="image pre-icon tooltip"
    fill="BlackCommon"
    height="17"
    width="17"
  >
    <use
      xlink:href="test-file-stub#key"
    />
  </svg>
  <input
    class="input"
    id="test-id"
    type="text"
  />
</div>
`;
